app/src/main/java/com/example/androidfundamental/retrofitexample/ui/RetrofitMainActivity.kt

```diff
@@ -0,0 +1,169 @@
+package com.example.androidfundamental.retrofitexample.ui
+
+import android.app.Activity
+import android.content.Intent
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.util.Log
+import android.view.View
+import android.widget.*
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import com.example.androidfundamental.R
+import com.example.androidfundamental.retrofitexample.adapter.RecyclerViewAdapter
+import com.example.androidfundamental.retrofitexample.api.StackOverflowAPI
+import com.example.androidfundamental.retrofitexample.model.Answer
+import com.example.androidfundamental.retrofitexample.model.ListWrapper
+import com.example.androidfundamental.retrofitexample.model.Question
+import com.google.gson.GsonBuilder
+import okhttp3.ResponseBody
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+
+class RetrofitMainActivity : Activity(), View.OnClickListener {
+    private var stackoverflowAPI: StackOverflowAPI? = null
+    private var token: String? = null
+    private var authenticateButton: Button? = null
+    private var questionsSpinner: Spinner? = null
+    private var recyclerView: RecyclerView? = null
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.retrofit_activity_main)
+        questionsSpinner = findViewById<View>(R.id.questions_spinner) as Spinner
+        questionsSpinner!!.onItemSelectedListener = object : AdapterView.OnItemSelectedListener {
```

Comment:
Whenever you notice that one view will be created in onCreate you should declared as lateinit, in that way you'll avoid the double bang "!!"